### ISSUE TITLE
New LSVE SwissTXT streams

### ISSUE THREAD

**Issue by defagos** [SRGSSR/srgcontentprotection-apple#6]
We must test the new SwissTXT LSVE streams. In particular, we must also check that they can be played with an appropriate token.

**Comment by pyby** [comment]
The same issue appears on the new RTS VoD service too. In this case, Akamai is the CDN but not the packaging service. It's a nginx service hosted by SRG SSR, and not protected by an Akamai token.
https://rts-vod-amd.akamaized.net/ww/hls/10623665/bf3e6e71-420c-307d-9a03-399209066fd9/master.m3u8

After investigation, I founded why the new LSVE test stream does not work with our current implementation: the master playlist uses relative URLs for child playlists, which is not the case on all other Akamai urls, used at SRG SSR.

Here is the use case:
The `-assetURLForURL:` changes the scheme on the first resource loading request, like:
`akamai+https://[host]/[path_to_stream]/master.m3u8` (it's not necessary to end with `master.m3u8`, it's for the example).

Then, `AVPlayer` detects an unknown resource and call:
`-shouldProcessResourceLoadingRequest:`. Our implementation does:
1. Restore the original scheme.
2. Call SRG SSR token service.
3. Modify the url request with the given token.
4. Load the request content and send the body to the resource loading request.

For all master playlists from Akamai used by SRG SSR: each video, audio, or subtitles variant tracks has a full URL, like:
`https://[host]/[path_to_stream]/child_playlist_q180.m3u8` (it's not necessary to end with `.m3u8`, it's for example).

This new LSVE master playlist has relative urls for variant tracks, like:
`child_playlist_q180.m3u8` (it's not necessary to end with `.m3u8`, it's for example).

The HLS specification allows to use relative urls, but the issue in our implementation:

- With full URLs, only the master playlist is modified, tokenised, and loaded within the code, all other HLS requests (keys, child playlists, chunks…) have the `https` scheme and `-shouldProcessResourceLoadingRequest:` is not called anymore.
- With relative URLs, `AVPlayer` has the first resource loading request from the URL created by `-assetURLForURL:`, which is `akamai+https://[host]/[path_to_stream]/master.m3u8`. The next URLs are relatives, so that the player reuses the previous request. All URLs have the `akamai+https` scheme and `-shouldProcessResourceLoadingRequest:` is called for each HLS resource. Our implementation does not expect to load child playlist or chunk media files. The SRGSSR token service is also call each time… The player stops after few request tries, and unexpected loading response and body (I don't know why).

To support our current Akamai token implementation, a stream packaging provider has to deliver full URLs in HLS playlists, master playlist included.

To test our self, I created `feature/lsve-tests` branch.
